### ISSUE TITLE
Let block issues for the generated functions

### ISSUE THREAD

**Issue by thautwarm** [JuliaLang/julia#33414]
# Version Info

```julia
julia> versioninfo()
Julia Version 1.2.0
Commit c6da87ff4b (2019-08-20 00:03 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.1 (ORCJIT, skylake)

sh> uname -a
Linux $USER 4.19.69-1-MANJARO #1 SMP PREEMPT Thu Aug 29 08:51:46 UTC 2019 x86_64 GNU/Linux
```

# Reproduction 

## Code

```julia
@generated function f(x::T)  where T
           quote
               let T = Core.Box(T)
                   G = T.contents
                   T.contents = String
                   let tp = (G, T)
                       tp
                   end
               end
           end
       end
f(1)
```

## Expected

```julia
(Int64, Core.Box(String))
```
## Got

```julia
(Int64, Int64)
```

**Comment by Keno** [comment]
This is fixed for me on master and crashes 1.2. Can you check master to see if you original issue (assuming this is a reduction) is resolved there?

**Comment by brenhinkeller** [comment]
Looks like this is still reliably fixed as of 1.8.3 and 1.9.0-alpha1. Do we want a test before closing?
